actionpack/test/dispatch/middleware_stack_test.rb

```diff
@@ -213,4 +214,55 @@ def test_delete_works
   test "includes a middleware" do
     assert_equal true, @stack.include?(ActionDispatch::MiddlewareStack::Middleware.new(BarMiddleware, nil, nil))
   end
+
+  test "can add another middleware stack to a stack" do
```

Comment:
Let's move this test to the `MiddlewareStackProxy` tests. This way we avoid making actionpack know about the existence of that class.

Reply:
I've moved the flattening logic to the `MiddlewareStackProxy` class and moved the corresponding tests.

If this seems reasonable I'll squash the branch.